### PR TITLE
KAFKA-3847: Use a separate producer per source task

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@hachikuji @Ishiihara @shikhar @ijuma @gwenshap

**Comment by hachikuji** [comment]
LGTM. Wow, that was even less work than I was expecting.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public void initialize(TaskConfig taskConfig) {
     }
 
     protected void close() {
-        // nothing to do
+        producer.close();
```
Comment (hachikuji) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java]: One question: would it make sense to use a timeout when closing the producer? Or would we have already flushed by this time?

Comment (ewencp) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java]: I was thinking about the same thing. The Connect code actually _never_ calls `flush`. I've added the timeout. The choice of 30s was mostly arbitrary, but I don't want to expose a ton of configs just for timeouts like this.

**Comment by gwenshap** [comment]
LGTM. +1 for the decision not to add config from the timeout.

**Pull request closed without merging**